src/SeleniumLibrary/keywords/alert.py

```diff
@@ -14,152 +14,183 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import time
-
 from selenium.common.exceptions import WebDriverException
 from selenium.webdriver.support import expected_conditions as EC
 from selenium.webdriver.support.ui import WebDriverWait
 
-from SeleniumLibrary.base import keyword
-from SeleniumLibrary.base import LibraryComponent
+from SeleniumLibrary.base import keyword, LibraryComponent
 from SeleniumLibrary.utils import is_truthy
 
 
 class AlertKeywords(LibraryComponent):
+    ACCEPT = 'ACCEPT'
+    DISMISS = 'DISMISS'
+    LEAVE = 'LEAVE'
+    _next_alert_action = ACCEPT
 
-    ACCEPT_ALERT = 'accept'
-    DISMISS_ALERT = 'dismiss'
+    @keyword
+    def input_text_into_prompt(self, text):
+        """Deprecated. Use `Input Text Into Alert` instead.
 
-    def __init__(self, ctx):
-        LibraryComponent.__init__(self, ctx)
-        self._next_alert_dismiss_type = self.ACCEPT_ALERT
+        Types the given ``text`` into an input field in an alert.
+        Leaves the alert open.
+        """
+        self.input_text_into_alert(text, self.LEAVE)
 
     @keyword
-    def input_text_into_prompt(self, text):
-        """Types the given `text` into alert box.  """
-        try:
-            alert = self._wait_alert()
-            alert.send_keys(text)
-        except WebDriverException:
-            raise RuntimeError('There were no alerts')
+    def input_text_into_alert(self, text, action=ACCEPT):
+        """Types the given ``text`` into an input field in an alert.
+
+        The alert is accepted by default, but that behavior can be controlled
+        by using the ``action`` argument same way as with `Handle Alert`.
+
+        New in SeleniumLibrary 3.0.
+        """
+        alert = self._wait_alert()
+        alert.send_keys(text)
+        self._handle_alert(alert, action)
+
+    @keyword
+    def alert_should_be_present(self, text='', action=ACCEPT):
+        """Verifies that an alert is present and, by default, accepts it.
+
+        Fails if no alert is present. If ``text`` is a non-empty string,
+        then it is used to verify alert's message. The alert is accepted
+        by default, but that behavior can be controlled by using the
+        ``action`` argument same way as with `Handle Alert`.
+
+        The ``action`` argument is new in SeleniumLibrary 3.0. In earlier
+        versions the alert was always accepted.
+        """
+        message = self.handle_alert(action)
+        if text and text != message:
```

Comment:
Now the text is exact match, but I was thinking could `if text and text not in message:` be also useful. Just a random thought, without a real use case.

Reply:
Yeah, if I had designed this I'd probably supported glob-style patterns (e.g. `foo *`) similarly as `Run Keyword And Expect Error` does. Don't think changing existing logic is a good idea, though. Could add `pattern=False`, but that's a topic for a separate issue. It's anyway already possible to do, for example:

    ${text} =    Alert Should Be Present
    Should Match    ${text}    foo *

Reply:
True. And alerts are pretty simple in any case.